packages/fxa-settings/src/models/reliers/rellier.ts

```diff
@@ -0,0 +1,109 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import "reflect-metadata";
+import { ContextValidation as V, bind, validateContext, ContextKeyTransforms as T, ContextProvider } from '../../lib/context';
+import { Context } from "../../lib/context/interfaces/context";
+
+/**
+ * A convenience interface for easy mocking / testing. Use this type in components.
+ */
+export interface RelierData
+{
+  context: string;
+  name: string;
+  email: string | undefined;
+  entrypoint: string | undefined;
+  entrypointExperiment: string | undefined;
+  entrypointVariation: string | undefined;
+  resetPasswordConfirm: boolean | undefined;
+  setting: string | undefined;
+  service: string | undefined;
+  serviceName: string | undefined;
+  subscriptionProductId: string | undefined;
+  subscriptionProductName: string | undefined;
+  style: string | undefined;
+  uid: string | undefined;
+  utmCampaign: string | undefined;
+  utmContent: string | undefined;
+  utmMedium: string | undefined;
+  utmSource: string | undefined;
+  utmTerm: string | undefined;
+}
+
+/**
+ * Create a relier class that can be bound to a given context.
+ */
+export class Relier implements ContextProvider, RelierData {
+
+  @bind([V.isString, V.isRequired])
+  context: string = '';
+
+  @bind([V.isString])
+  name: string = '';
+
+  @bind([V.isString])
+  email: string | undefined;
+
+  @bind([V.isString])
+  entrypoint: string | undefined;
+
+  @bind([V.isString], T.snakeCase)
+  entrypointExperiment: string | undefined;
+
+  @bind([V.isString], T.snakeCase)
+  entrypointVariation: string | undefined;
+
+  @bind([V.isString], T.snakeCase)
+  resetPasswordConfirm: boolean | undefined;
+
+  @bind([V.isString], "setting")
+  setting: string | undefined;
+
+  @bind([V.isString], "service")
+  service: string | undefined;
+
+  // TODO: Double check
+  @bind([V.isString], "service")
+  serviceName: string | undefined;
```

Comment:
FWIW we started an enum called `MozServices` that we'll add to as we confirm service names during development. Not sure if it should be used here or not since it's incomplete though.

Also, not sure if it's possible to reuse the types from `RelierData` but that could be nice. 🤔